src/main/kotlin/org/neo4j/graphql/Predicates.kt

```diff
@@ -4,22 +4,33 @@ import graphql.Scalars
 import graphql.schema.*
 import org.neo4j.graphql.Predicate.Companion.resolvePredicate
 import org.neo4j.graphql.handler.projection.ProjectionBase
+import org.neo4j.opencypherdsl.Condition
+import org.neo4j.opencypherdsl.Expression
+import org.slf4j.LoggerFactory
 
 interface Predicate {
     fun toExpression(variable: String): Cypher
 
     companion object {
         fun resolvePredicate(name: String, value: Any?, type: GraphQLFieldsContainer): Predicate {
-            val (fieldName, op) = Operators.resolve(name, value)
-            return if (type.hasRelationship(fieldName)) {
-                when (value) {
-                    is Map<*, *> -> RelationPredicate(fieldName, op, value, type)
-                    null -> IsNullPredicate(fieldName, op, type)
-                    else -> throw IllegalArgumentException("Input for $fieldName must be an filter-InputType")
+            for (definedField in type.fieldDefinitions) {
```

Comment:
looks more expensive, why is this necessary to iterate over all fields?

Reply:
We use different resolve logic for relations than for fields, so we need to know which logic to take